app/scripts/modules/core/delivery/service/execution.service.js

```diff
@@ -260,6 +260,18 @@ module.exports = angular.module('spinnaker.core.delivery.executions.service', [
       }
     }
 
+    function getLastExecution(appName, configId) {
+      return getExecutions(appName)
+        .then((executions) => {
+          return executions.filter((execution) => {
+            return execution.pipelineConfigId === configId;
+          });
+        })
+        .then((executionsByConfigId) => {
+          return executionsByConfigId;
+        });
```

Comment:
Doesn't look like this `.then` is doing anything, but could be tweaked to return just the first item (or null).